server/chat.ts

```diff
@@ -1141,6 +1116,55 @@ export class CommandContext extends MessageContext {
 
 		return message;
 	}
+	canPM(targetUser: User) {
+		// hotpatching
+		const user = this.user;
+		if (typeof user.settings.blockPMs === 'boolean') {
+			user.settings.blockPMs = {all: user.settings.blockPMs};
+		}
+		if (typeof targetUser.settings.blockPMs === 'boolean') {
+			targetUser.settings.blockPMs = {all: targetUser.settings.blockPMs};
+		}
```

Comment:
You have this code in five different places and I can't help but think there's a better way.

At least make it a `Chat.upgradeSettings` function?